lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -1132,6 +1132,10 @@ impl HNSWIndex {
         params: Option<&SearchParams>,
         vector_query_context: &VectorQueryContext,
     ) -> OperationResult<Vec<Vec<ScoredPointOffset>>> {
+        if top == 0 {
```

Comment:
Do we also need this check for the indexed pass?

Reply:
Thanks for the test, I like your decision about moving the check to the upper call stack